### PR TITLE
Fix Out-of-Band changes to service_connect_configuration dont cause drift

### LINKED ISSUE #42818: aws_ecs_service - changes to service connect configuration are not detected
### Terraform and AWS Provider Version

```console
Terraform v1.12.1
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v5.99.0
+ provider registry.terraform.io/hashicorp/random v3.7.2
```

### Affected Resource(s) or Data Source(s)

* aws_ecs_service

### Expected Behavior

changes to service connect configuration should be detected by terraform

### Actual Behavior

changes are not detected

### Sample Terraform Configuration

I don't have readily available config to provide.

### Steps to Reproduce

1.  Create an ECS service with service connect configuration defined
2. verify in the console that it is present, or in some cases like mine terraform will successfully apply but the console shows service connect is turned off. if it is turned on, just manually disable it in the console
3. run terraform plan, you'll see no changes

### GenAI / LLM Assisted Development

n/a

### Important Facts and References

supposedly was solved here, but recent testing indicates it is not solved. 

https://github.com/hashicorp/terraform-provider-aws/issues/40715

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by dani-chmb**
## Rollback Plan
If a change needs to be reverted, we will publish an updated version of the library.
## Changes to Security Controls
N/A
### Description

[This issue](https://github.com/hashicorp/terraform-provider-aws/issues/42818) highlights a bug with the `service_connect_configuration` (SCC) property of `aws_ecs_service` resource.

**Expected Behavior**: For an ECS Service managed by Terraform, initially created with SCC enabled, Terraform should detect whenever SCC is disabled Out-of-Band. The next `terraform plan` should show the drift.
**Actual Behavior**: After the Out-of-Band update, `terraform plan` shows "No changes" even when Service Connect configuration has been manually modified in the AWS Console.

Terraform invokes the DescribeServices API as part of `resourceServiceRead`, which is invoked by `terraform plan` . When SCC is Enabled, `DescribeServices` returns a `service_connect_configuration` as part of its response. However, if it's Disabled, there's no mention of SCC on the response. In the current logic, the provider does not update State if  SCC is `null` on response, hence the "No Changes Detected".

This change explicitly sets SCC to `null` in Terraform State if it's also `null` in the `DescribeServices` response. Now, when we remove SCC Out-of-Band and follow-up with `terraform plan`, terraform should flag the `null` SCC as "Changes Needed".
### Relations

Closes #42818
### References
* I modeled the Out of Band Acceptance test (as well as changes to helper functions) after this [IAM Role Out of Band Test](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/iam/role_test.go#L747).
* A similar issue #40715 was previously addressed in PR #41998 , but the Out-of-Band edge case still should be patched.

### Output from Acceptance Testing

**New test**: `TestAccECSService_ServiceConnect_outOfBandRemoval`
```console
% make testacc TESTS=TestAccECSService_ServiceConnect_outOfBandRemoval PKG=ecs
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.24.5 test ./internal/service/ecs/... -v -count 1 -parallel 20 -run='TestAccECSService_ServiceConnect_outOfBandRemoval'  -timeout 360m -vet=off
2025/08/13 14:49:05 Creating Terraform AWS Provider (SDKv2-style)...
2025/08/13 14:49:05 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccECSService_ServiceConnect_outOfBandRemoval
=== PAUSE TestAccECSService_ServiceConnect_outOfBandRemoval
=== CONT  TestAccECSService_ServiceConnect_outOfBandRemoval
--- PASS: TestAccECSService_ServiceConnect_outOfBandRemoval (144.20s)
PASS
ok      [[github.com/hashicorp/terraform-provider-aws/internal/service/ecs](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs)](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs))        144.376s
...
```
**Existing Service Connect Tests**
```console
% make testacc TESTS=TestAccECSService_ServiceConnect_remove PKG=ecs
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.24.5 test ./internal/service/ecs/... -v -count 1 -parallel 20 -run='TestAccECSService_ServiceConnect_remove'  -timeout 360m -vet=off
2025/08/13 14:41:21 Creating Terraform AWS Provider (SDKv2-style)...
2025/08/13 14:41:21 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccECSService_ServiceConnect_remove
=== PAUSE TestAccECSService_ServiceConnect_remove
=== CONT  TestAccECSService_ServiceConnect_remove
--- PASS: TestAccECSService_ServiceConnect_remove (147.23s)
PASS
ok      [[github.com/hashicorp/terraform-provider-aws/internal/service/ecs](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs)](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs))        147.456s
```
```console
%  make testacc TESTS=TestAccECSService_ServiceConnect_basic PKG=ecs
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.24.5 test ./internal/service/ecs/... -v -count 1 -parallel 20 -run='TestAccECSService_ServiceConnect_basic'  -timeout 360m -vet=off
2025/08/13 14:41:21 Creating Terraform AWS Provider (SDKv2-style)...
2025/08/13 14:41:21 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccECSService_ServiceConnect_basic
=== PAUSE TestAccECSService_ServiceConnect_basic
=== CONT  TestAccECSService_ServiceConnect_basic
--- PASS: TestAccECSService_ServiceConnect_basic (161.30s)
PASS
ok      [[github.com/hashicorp/terraform-provider-aws/internal/service/ecs](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs)](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs](http://github.com/hashicorp/terraform-provider-aws/internal/service/ecs))        161.460s
```

### CODE REVIEW

File: internal/service/ecs/service_test.go [hunk 1]
```diff
@@ -2500,6 +2533,29 @@ func testAccCheckServiceExists(ctx context.Context, name string, service *awstyp
 			return nil
 		})
 
+		if err != nil {
+			return err
+		}
+
+		*service = *output
```
Comment (dani-chmb) [inline internal/service/ecs/service_test.go:2819]: This change allows us to reuse the data in the `service` struct for populating the `UpdateServiceInput` in the Out-of-band helper method `testAccCheckServiceDisableServiceConnect`. This logic is based on the equivalent method for `aws_iam_role` AccTests ([see link](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/iam/role_test.go#L1000))

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccECSService_ServiceConnect' PKG=ecs ACCTEST_PARALLELISM=3
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.24.6 test ./internal/service/ecs/... -v -count 1 -parallel 3  -run=TestAccECSService_ServiceConnect -timeout 360m -vet=off
2025/08/14 09:43:41 Creating Terraform AWS Provider (SDKv2-style)...
2025/08/14 09:43:41 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccECSService_ServiceConnect_basic
=== PAUSE TestAccECSService_ServiceConnect_basic
=== RUN   TestAccECSService_ServiceConnect_full
=== PAUSE TestAccECSService_ServiceConnect_full
=== RUN   TestAccECSService_ServiceConnect_tls_with_empty_timeout
=== PAUSE TestAccECSService_ServiceConnect_tls_with_empty_timeout
=== RUN   TestAccECSService_ServiceConnect_ingressPortOverride
=== PAUSE TestAccECSService_ServiceConnect_ingressPortOverride
=== RUN   TestAccECSService_ServiceConnect_remove
=== PAUSE TestAccECSService_ServiceConnect_remove
=== RUN   TestAccECSService_ServiceConnect_outOfBandRemoval
=== PAUSE TestAccECSService_ServiceConnect_outOfBandRemoval
=== CONT  TestAccECSService_ServiceConnect_ingressPortOverride
=== CONT  TestAccECSService_ServiceConnect_full
=== CONT  TestAccECSService_ServiceConnect_outOfBandRemoval
--- PASS: TestAccECSService_ServiceConnect_ingressPortOverride (168.59s)
=== CONT  TestAccECSService_ServiceConnect_tls_with_empty_timeout
--- PASS: TestAccECSService_ServiceConnect_full (168.61s)
=== CONT  TestAccECSService_ServiceConnect_remove
--- PASS: TestAccECSService_ServiceConnect_outOfBandRemoval (148.21s)
=== CONT  TestAccECSService_ServiceConnect_basic
--- PASS: TestAccECSService_ServiceConnect_remove (136.61s)
--- PASS: TestAccECSService_ServiceConnect_tls_with_empty_timeout (166.89s)
--- PASS: TestAccECSService_ServiceConnect_basic (152.08s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ecs	373.745s
```

**Review by jar-b — APPROVED** [review]
LGTM 🚀

**Comment by ewbankkit** [comment]
@dani-chmb Thanks for the contribution :tada: :clap:.
